### ISSUE TITLE
VS task runner uses hardcoded paths

### ISSUE THREAD

**Issue by shrayasr** [shrayasr/Iko#1]
`/u/opium_tm` says ([permalink](https://www.reddit.com/r/dotnet/comments/73cegg/a_cli_based_task_runner_written_in_c/do2safo/)):

> VS runner is implemented completely wrong way. Paths to VS are hardcoded in lib. For VS before 2017 you should get installation paths from well known registry branch. For VS 2017 you should use documented COM interfaces to enumerate (possibly side by side) VS 2017 installations.

Fair point. Everything in the project is in a "works-for-me" kind of situation.

Basically we're currently at:

- [X] ~~Make it work~~
- [ ] Make it right
- [ ] Make it fast

**Comment by SpartanX1** [comment]
Hello I am interested in solving this issue. Are you still active in this project?

**Comment by shrayasr** [comment]
Oh yes. I use it everyday :) Please go ahead.

**Comment by SpartanX1** [comment]
Hey you did not mention how to and where to place the config file?

**Comment by shrayasr** [comment]
Which config file @SpartanX1? The base iko config is placed at `~/iko.toml`

**Comment by SpartanX1** [comment]
Do i need to install toml or something ? or just create a file with this name

**Comment by SpartanX1** [comment]
I have made changes for vs17 ,  but i need to check how it works. 
Here's the syntax. I hope there is a space between path and @params? 

`var` path = GetVisualStudioInstallationFolder();

`Process.Start("cmd", $@"/c start "+path+" "+@params.Path);`

**Comment by SpartanX1** [comment]
I am not clear on how to run and test the application. 
Could you please provide the steps? @shrayasr

**Comment by shrayasr** [comment]
> Do i need to install toml or something ? or just create a file with this name

Not at all. TOML is a plain text file in a certain format. Just creating a file with the name is enough.

> I am not clear on how to run and test the application.
> Could you please provide the steps?

- Build it 
- `dotnet \path\to\iko.dll <task name>`

This assumes that you have already defined the tasks in the TOML file. In your case, the TOML file could be: 

```toml
[ikosln]
cmd = 'vs17'
path = 'c:\users\spartanx1\projects\iko\iko.sln'
```

and run it as

```
dotnet \path\to\iko.dll ikosln
```

**Comment by SpartanX1** [comment]
Thanks I will test it out

**Comment by shrayasr** [comment]
You're welcome. Thank you for working on this 🎉

**Comment by SpartanX1** [comment]
Hello could you review my pull request @shrayasr

**Comment by shrayasr** [comment]
Will do it soon, @SpartanX1. Held up at work at the moment

**Comment by SpartanX1** [comment]
Sure thanks:)

**Comment by aintabb** [comment]
@shrayasr Is this issue still open?
